frontend/src/pages/Leaderboard.js

```diff
@@ -145,90 +152,115 @@ const Leaderboard = () => {
+                    >
+                      Earnings
+                    </TableHeaderCell>
+                    <TableHeaderCell
+                        onClick={() => sortData('xp')}
+                        sortDirection={sortConfig.key === 'xp' ? sortConfig.direction : null}
+                    >
+                      XP
+                    </TableHeaderCell>
+                    <TableHeaderCell
+                        onClick={() => sortData('Features')}
+                        sortDirection={sortConfig.key === 'Features' ? sortConfig.direction : null}
+                    >
+                      Features
+                    </TableHeaderCell>
+                    <TableHeaderCell
+                        onClick={() => sortData('Bugs')}
+                        sortDirection={sortConfig.key === 'Bugs' ? sortConfig.direction : null}
+                    >
+                      Bugs
+                    </TableHeaderCell>
+                    <TableHeaderCell
+                        onClick={() => sortData('Optimisations')}
+                        sortDirection={sortConfig.key === 'Optimisations' ? sortConfig.direction : null}
+                    >
+                      Optimisations
+                    </TableHeaderCell>
+                  </TableRow>
+                </TableHeader>
+                <tbody>
+                {filteredData.map((user, index) => (
+                    <TableRow key={user._id}>
+                      <TableCell>{index + 1}</TableCell>
+                      <TableCell>
+                        <div style={{ display: 'flex', alignItems: 'center' }}>
+                          <Link to={`/profile/${user.username}`} style={{ color: 'white', textDecoration: 'none' }}>
+                            <ProfileImage
+                                src={user.avatar || 'https://img.freepik.com/free-psd/3d-render-avatar-character_23-2150611746.jpg'}
+                                alt="profile"
+                            />
+                          </Link>
+                          <Link to={`/profile/${user.username}`} style={{ color: 'white', textDecoration: 'none' }}>
+                            {user.username}
+                          </Link>
+                        </div>
+                      </TableCell>
+                      <TableCell>
+                        <EarningsWrapper>
+                          {user.totalEarnings}
+                          <EarningsLogo src={usdcLogoUrl} alt="USDC logo" />
+                        </EarningsWrapper>
+                      </TableCell>
+                      <TableCell>{user.xp}</TableCell>
+                      <TableCell>{user.Features}</TableCell>
+                      <TableCell>{user.Bugs}</TableCell>
+                      <TableCell>{user.Optimisations}</TableCell>
+                    </TableRow>
+                ))}
+                </tbody>
+              </Table>
+            </LeaderboardWrapper>
```

Comment:
Am sorry but this whole page has to be done better. Make the components smaller, that makes it reusable and readable.